webapp/channels/src/entry.tsx

```diff
@@ -77,3 +77,26 @@ appendOnDOMContentLoadedEvent(() => {
     // Do the pre-render setup and call renderReactRootComponent when done
     preRenderSetup(renderReactRootComponent);
 });
+
+const registerServiceWorker = async () => {
```

Comment:
We should probably have a feature flag or config setting to enable/disable this. While the service worker is likely fine for most people, I imagine this is going to cause issues with subpaths or any of the wonky setups people have where they host MM on the same domain as other static pages

Reply:
Can you clarify how this is going to cause issues with subpaths/separate static page?

Reply:
Because everything breaks subpaths randomly 😛

Jokes aside, I'm worried that we'll run into some situation where something is incorrectly cached and we're going to need to start telling people to clear their local data for the web app periodically like we have to with desktop or mobile, and that seems more likely for non-standard setups like subpaths.

Generally, I think we should add on-by-default settings as escape hatches for more changes like this to make sure that we have a fallback if anything goes wrong to avoid admins having to downgrade their servers or anything